jena-base/src/main/java/org/apache/jena/atlas/lib/FileOps.java

```diff
@@ -63,12 +63,19 @@ public static void delete(File f, boolean reportExistsAfter) {
         }
     }
 
-    /** Delete all files in a directory */
+    /**
+     * Delete all files in a directory.
+     * Does nothing if the path name does not exist or is not a directory.
+     *
+     * @param dir
+     */
     public static void clearDirectory(String dir) {
         File d = new File(dir) ;
-        for ( File f : d.listFiles() ) {
-            if ( f.isFile() )
-                delete(f, false) ;
+        if ( d.isDirectory() ) {
+            for ( File f : d.listFiles() ) {
+                if ( f.isFile() )
```

Comment:
I take it this never needs to recurse into further levels of directory?

Reply:
This one only deletes files. There's a separate `clearAll` for recursive deletion.

It gets used mostly in testing (so, an expected controlled environment) and limits the damage in case things aren't as expected.